Documentation/security/threat-model.rst

```diff
@@ -0,0 +1,751 @@
+|                  |    pod identities.                                |
+|                  |                                                   |
+|                  | Denial of service could occur depending on the    |
+|                  | behavior of the attacker.                         |
++------------------+---------------------------------------------------+
+| Observability    | Denial of service could occur depending on the    |
+| data             | behavior of the attacker.                         |
+|                  | |br| |br|                                         |
+|                  | Information leakage could occur via an attacker   |
+|                  | scraping Cilium or Hubble Prometheus metrics,     |
+|                  | depending on the specific metrics enabled.        |
++------------------+---------------------------------------------------+
+
+Recommended Controls
+^^^^^^^^^^^^^^^^^^^^
+
+:ref:`gsg_encryption` should be configured to ensure the confidentiality of
+communication between workloads.
+
+.. _kubernetes-api-server-attacker:
+
+Kubernetes API Server Attacker
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This type of attack could be carried out by any user or code with
+network access to the Kubernetes API server and credentials that allow
+Kubernetes API requests. Such permissions would allow the user to read
+or manipulate the API server state (for example by changing CRDs).
+
+This is not a precise term, as different RBAC policies could grant many
+different types of privilege. This section is intended to cover any
+attack that might be exposed via Kubernetes API server access, whether
+it is “full” API server access, or even limited API server access (for
+example, access gained by compromising credentials used by Cilium).
+
+`The Kubernetes documentation`_ offers more information about configuring RBAC.
+
+.. _The Kubernetes documentation: https://kubernetes.io/docs/reference/access-authn-authz/rbac/
+
+For such an attacker, our threat matrix looks like:
+
+.. rst-class:: wrapped-table
+
++------------------+---------------------------------------------------+
+| **Threat         | **Identified STRIDE threats**                     |
+| surface**        |                                                   |
++==================+===================================================+
+| Cilium agent     | A Kubernetes API user with ``kubectl exec``       |
+|                  | access to the pod running Cilium effectively      |
+|                  | becomes a :ref:`root-equivalent host              |
+|                  | attacker <root-equivalent-host-attacker>`,        |
+|                  | since Cilium runs as a privileged pod.            |
+|                  | |br| |br|                                         |
+|                  | Users with permissions to configure workload      |
+|                  | settings could cause denial of service.           |
```

Comment:
I think that this one is similar to the workload attacker entry, including the Cilium security benefits? Do you think there is a good way to cross-reference?